### PR TITLE
[FIX] Adding data type support  for variable replacement

### PR DESCRIPTION
**Pull request by harini-venkataraman**
## What
PR adds replacement support for JSON and data types other than string for Variable.
Also, a few loggers is added for better interpretation of replacement.
...

## Why
In variable extraction, in order to support JSON type responses from the POST API response.
...

## How
Support using instance type received. 
...

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)
No, this adds enhancements to the Variable replacement feature
...

## Database Migrations
Not involved.
...

## Env Config
Not involved.
...

## Relevant Docs
Not involved.
...

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py [hunk 1]
```diff
@@ -44,10 +45,21 @@ def check_static_variable_run_status(
             return output
 
     @staticmethod
-    def replace_generic_string_value(prompt: str, variable: str, value: str) -> str:
-        replaced_prompt = prompt.replace(variable, value)
+    def replace_generic_string_value(prompt: str, variable: str, value: Any) -> str:
+        formatted_value: str = value
+        if not isinstance(value, str):
+            formatted_value = VariableService.handle_json_and_str_types(value)
```
Comment (chandrasekharan-zipstack) [inline prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py:51]: NIT: We are checking if its not an instance of str type but we end up calling a function named `VariableService.handle_json_and_str_types()` which doesn't help improve the understanding without stepping into the code. Try to improve the naming maybe

File: prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py [hunk 2]
```diff
@@ -44,10 +45,21 @@ def check_static_variable_run_status(
...
+        replaced_prompt = prompt.replace(variable, formatted_value)
         return replaced_prompt
 
+    @staticmethod
+    def handle_json_and_str_types(value: Any) -> str:
+        try:
+            formatted_value = json.dumps(value)
+        except ValueError:
```
Comment (chandrasekharan-zipstack) [inline prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py:59]: @harini-venkataraman shouldn't this be `JSONDecodeError`?

**Review by chandrasekharan-zipstack — APPROVED** [review]

File: prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py (hunk 1 above)
Comment (harini-venkataraman) [reply prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py:51]: @chandrasekharan-zipstack Noted the suggestion. Will pick it up in later optimizations.

File: prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py (hunk 2 above)
Comment (harini-venkataraman) [reply prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py:59]: Yes @chandrasekharan-zipstack Like we discussed, ValueError is parent of `JSONDecodeError`, Owing to priority, let us merge this. Will pick it up in next PR.

**Review by Deepak-Kesavan — APPROVED** [review]

**Pull request merged**